### DISCUSSION TITLE
[DONE] Prevent editing the same page/collection element

### DISCUSSION THREAD

**Discussion by netzpixel** [payloadcms/payload#10124] [Feature Requests & Ideas]
I'm searching for a new headless CMS and in my search I came across payload. I like the concept of it etc. but it seams to have the same problem as the CMS I'm switching from. That why I'm asking if this feature is actually missing or if I just didn't find it?

When multiple users edit - for example - the same page and save it, the last one saving "wins" and all other data saved by another user is "lost" (not really it's available in the versions but it's not merged or anything else).
My feature request would be very simple: Either lock element which are being edited by other users OR somehow merge the edited data (at least the data from field which have not been touched by the user saving last).

**Comment by jacobsfletch** [comment]
As of v3, Payload locks documents to a single user by default. More on this here: https://payloadcms.com/docs/admin/locked-documents

**Reply by netzpixel** [reply]
Thanks for the quick response. In that case there might be a bug. 
I used `npx create-payload-app@latest` to create a new project with the template and all I did was:
* creating a page
* creating a second user
* start editing the pages hero with user one
* doing the same with user two
* saving with both - the last info was saved, the first was pushed into a version

So what did I do wrong?

**Reply by netzpixel** [reply]
I just tested a bit more and it seams that there is some strange timing issue. It now works as it should most of the time but some times, when both start editing at the same time, there is no info about someone else already locking the element. But I think that's a none issue, so this can be closed.
